common/src/Microsoft.Azure.IIoT.Hub.Module.Framework/src/Hosting/MethodRouter.cs

```diff
@@ -300,8 +300,9 @@ private class DefaultFilter : ExceptionFilterAttribute {
                         }
                         _logger.Verbose(ex, "Method call error");
                         ex = _ef.Filter(ex, out var status);
+
                         if (ex is MethodCallStatusException) {
-                            throw ex;
+                            throw new MethodCallStatusException(ex.Message, status);
```

Comment:
Same comment for changes on line `331` below.

Reply:
The exception response is not the same.
`throw new MethodCallStatusException(ex.Message, status);` generates:
`{
    "status":404,
    "payload":{
        "Message":"Response 404 Nodes not found",
        "Description":{
            "EndpointUrl":"opc.tcp://192.168.1.70:50000",
            "UseSecurity":false,
            "OpcAuthenticationMode":"Anonymous",
            "OpcNodes":[
                {
                    "Id":"ns=2;s=SlowUInt1",
                    "OpcPublishingInterval":1000
                }
            ]
        }
    }
}
`
while `throw ex` generates:
`{
    "status":404,
    "payload":{
        "EndpointUrl":"opc.tcp://192.168.1.70:50000",
        "UseSecurity":false,
        "OpcAuthenticationMode":"Anonymous",
        "OpcNodes":[
            {
                "Id":"ns=2;s=SlowUInt1",
                "OpcPublishingInterval":10000
            }
        ]
    }
}
`

Reply:
@dacolgit, is this the result of direct method invocation ?

Reply:
I believe that this difference is caused by the fact that here you are transferring the value that has bees stored in `Message` to `ResponsePayload`, effectively overwriting it. And `ResponsePayload` is what is being picked up by `MethodRouter` when assembling the response to a direct method call. This makes me realize that we are incorrectly assigning assembled JSON to the `Message` field instead of `ResponsePayload` in the exception. So I think that we need to change the constructors that we have to contain the JSON message in `ResponsePayload`.

So this:

```csharp
        public MethodCallStatusException(int result, string errorMessage = null) :
            this("{}", result, errorMessage) {
            ResponsePayload = $"\"{errorMessage}\"";
        }

        ...
        public MethodCallStatusException(string responsePayload, int result,
            string errorMessage = null) :
            base($"{{\"Message\":\"Response {result} {errorMessage ?? ""}\",\"Description\":{responsePayload}}}") {
            Result = result;
            ResponsePayload = responsePayload;
        }
```

Should become something like this:

```csharp
        public MethodCallStatusException(int result, string errorMessage) :
            this("{}", result, errorMessage) { }

        ...
        public MethodCallStatusException(string responsePayload, int result, string errorMessage = null) :
            base($"Response {result} {errorMessage ?? ""}") {
            Result = result;
            ResponsePayload = $"{{\"Message\":\"Response {result} {errorMessage ?? ""}\",\"Description\":{responsePayload ?? "{}"}}}";
        }
```

Reply:
@dacolgit, is this the result of direct method invocation ?
Yes